crates/protect-ffi/src/lib.rs

```diff
@@ -169,105 +218,69 @@ async fn new_client_inner(
 
     let cipher = ScopedZeroKMSNoRefresh::init(zerokms.clone(), None).await?;
 
-    Ok(Client {
+    let client = Client {
         cipher: Arc::new(cipher),
         zerokms,
-        encrypt_config: Arc::new(encrypt_config.into_config_map()),
-    })
-}
+        encrypt_config: Arc::new(opts.encrypt_config.into_config_map()),
+    };
 
-fn encrypt(mut cx: FunctionContext) -> JsResult<JsPromise> {
-    let client = (**cx.argument::<JsBox<Client>>(0)?).clone();
-    let (plaintext_target, ident) = plaintext_target_from_js_object(
-        cx.argument::<JsObject>(1)?,
-        &client.encrypt_config,
-        &mut cx,
-    )?;
-    let service_token = service_token_from_js_value(cx.argument_opt(2), &mut cx)?;
-
-    let rt = runtime(&mut cx)?;
-    let channel = cx.channel();
-
-    // Create a JavaScript promise and a `deferred` handle for resolving it.
-    // It is important to be careful not to perform failable actions after
-    // creating the promise to avoid an unhandled rejection.
-    let (deferred, promise) = cx.promise();
-
-    // Spawn an `async` task on the tokio runtime. Only Rust types that are
-    // `Send` may be moved into this block. `Context` may not be passed and all
-    // JavaScript values must first be converted to Rust types.
-    //
-    // This task will _not_ block the JavaScript main thread.
-    rt.spawn(async move {
-        let ciphertext_result = encrypt_inner(client, plaintext_target, ident, service_token).await;
-
-        // Settle the promise from the result of a closure. JavaScript exceptions
-        // will be converted to a Promise rejection.
-        //
-        // This closure will execute on the JavaScript main thread. It should be
-        // limited to converting Rust types to JavaScript values. Expensive operations
-        // should be performed outside of it.
-        deferred.settle_with(&channel, move |mut cx| {
-            let ciphertext = ciphertext_result.or_else(|err| cx.throw_error(err.to_string()))?;
-            eql_encrypted_to_js(ciphertext, &mut cx)
-        });
-    });
-
-    Ok(promise)
+    Ok(Boxed(client))
 }
 
-async fn encrypt_inner(
-    client: Client,
-    plaintext_target: PlaintextTarget,
-    ident: Identifier,
-    service_token: Option<ServiceToken>,
-) -> Result<Encrypted, Error> {
+#[neon::export]
+async fn encrypt(
```

Comment:
The general pattern for exported functions is now:
- pass in the boxed client as the first arg
- all other args are wrapped in an opts object that implements `Serialize`/`Deserialize`

We split the two because the boxed client isn't really serializable as JSON.